sim/battle.ts

```diff
@@ -282,6 +282,13 @@ export class Battle {
 		this.prng = new PRNG(this.prng.startingSeed);
 	}
 
+	getVolatile(id: string): Effect {
+		if (id.startsWith('move:')) return Object.assign(Object.create(this.dex.getMove(id.slice(5))), {id});
```

Comment:
This also still seems wrong, `id` for Effect should be an `ID`.

Reply:
This was just bad naming.